### PR TITLE
No dashes in usernames

### PR DESCRIPTION
**Pull request by gabriel-v**
Hypothesis doesn't support dashes in its usernames.

### CODE REVIEW

File: liquidcore/home/models.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+from django.contrib.auth.models import AbstractUser
+
+USERNAME_CHARS = string.ascii_letters + string.digits + '.'
+
+
+class User(AbstractUser):
+    class Meta:
+        db_table = 'auth_user'
+
+    def save(self, *args, **kwargs):
```
Comment (mgax) [inline liquidcore/home/models.py]: Do we really have to change the user model? Is there no easier way to enforce this? Maybe as a pre-save hook?

File: liquidcore/home/templates/registration/login.html [hunk 2]
```diff
@@ -57,8 +59,14 @@ <h1>Login</h1>
...
-{% if not form.otp_token %}
-<p><a href="{% url 'password_reset' %}">Lost password?</a></p>
-{% endif %}
+<footer class="page-footer font-small blue pt-4">
+
+  {% if not form.otp_token %}
+  <p><a href="{% url 'password_reset' %}">Lost password?</a></p>
+  {% endif %}
+
+  <i>Note</i> All hyphens (<pre>-</pre>) in usernames have been replaced with dots (<pre>.</pre>).
```
Comment (mgax) [inline liquidcore/home/templates/registration/login.html]: Ugh, let's not keep this around forever. It's just 3 users, much easier to tell them directly.

**Review by mgax — CHANGES_REQUESTED** [review]
This introduces more complexity than it's worth, imho.

File: liquidcore/home/templates/registration/login.html (hunk 2 above)
Comment (gabriel-v) [reply liquidcore/home/templates/registration/login.html]: Ok, I'll kill this.

File: liquidcore/home/models.py (hunk 1 above)
Comment (gabriel-v) [reply liquidcore/home/models.py]: On a second look through the docs,

> [All built-in signals are sent using the send() method.](https://docs.djangoproject.com/en/2.2/ref/signals/)

> [send() does not catch any exceptions raised by receivers; it simply allows errors to propagate](https://docs.djangoproject.com/en/2.2/topics/signals/#django.dispatch.Signal.send)

Yes, we should be able to throw in the signal handler and expect the save will be aborted.

**Review by mgax — COMMENTED** [review]
I'm pushing my luck here, but can we have a unit test? It could be as simple as creating users via the user model and asserting the assertion error.

**Pull request merged**